libs/ledger-live-common/src/exchange/swap/hooks/useSwapTransaction.ts

```diff
@@ -172,7 +172,17 @@ export const useSwapTransaction = ({
       targetAccounts,
     },
     setFromAmount,
-    toggleMax,
+    toggleMax: () => {
+      /**
+       * When toggle max is triggered, we should also update `useAllAmount`
+       * This `useAllAmount` field should be standard / mandatory
+       */
+      bridgeTransaction.updateTransaction((tx) => {
+        tx.useAllAmount = !isMaxEnabled
+        return tx;
+      });
+      toggleMax();
```

Comment:
I think we are using 2 contradictories logic here:

useUpdateMaxAmount and your logic
```
  const toggleMax: SwapTransactionType["toggleMax"] = useCallback(
    () =>
    tx.useAllAmount = !isMaxEnabled(previous => {
        if (previous) {
          setFromAmount(ZERO);
          setIsMaxLoading(false);
        }
        return !previous;
      }),
    [setFromAmount],
  );
```